src/cache.ts

```diff
@@ -140,12 +144,12 @@ export class Cache {
         } else {
           meta$ = this.refreshMeta(qualified, cached.etag)
             .then((m) => {
-              cached.timestamp = Date.now();
               this.pendingRefreshes.del(qualified);
               if (m !== null) {
                 this.cache.set(qualified, m);
                 return m.meta;
               }
+              cached.timestamp = Date.now();
```

Comment:
no longer updating the timestamp when m!== null ? seems strange

Reply:
Maybe got it wrong, aren't we effectively storing a new cached value on `this.cache.set(qualified, m)`?
Couldn't see a reason to update the timestamp on `cached` there

Reply:
I see, missed that. There may be a case a caller was stuck on `await cached$` above and will get the previous version when resuming, but I'm not sure it can happen as the cached$ promise should be resolved in that case, and no context should block on it ... LGTM